provisioner/chef-solo/provisioner.go

```diff
@@ -45,26 +47,27 @@ var guestOSTypeConfigs = map[string]guestOSTypeConfig{
 type Config struct {
 	common.PackerConfig `mapstructure:",squash"`
 
-	ChefEnvironment            string   `mapstructure:"chef_environment"`
-	ChefLicense                string   `mapstructure:"chef_license"`
-	ConfigTemplate             string   `mapstructure:"config_template"`
-	CookbookPaths              []string `mapstructure:"cookbook_paths"`
-	RolesPath                  string   `mapstructure:"roles_path"`
-	DataBagsPath               string   `mapstructure:"data_bags_path"`
-	EncryptedDataBagSecretPath string   `mapstructure:"encrypted_data_bag_secret_path"`
-	EnvironmentsPath           string   `mapstructure:"environments_path"`
-	ExecuteCommand             string   `mapstructure:"execute_command"`
-	InstallCommand             string   `mapstructure:"install_command"`
-	RemoteCookbookPaths        []string `mapstructure:"remote_cookbook_paths"`
-	Json                       map[string]interface{}
-	PreventSudo                bool     `mapstructure:"prevent_sudo"`
-	RunList                    []string `mapstructure:"run_list"`
-	SkipInstall                bool     `mapstructure:"skip_install"`
-	StagingDir                 string   `mapstructure:"staging_directory"`
-	GuestOSType                string   `mapstructure:"guest_os_type"`
-	Version                    string   `mapstructure:"version"`
-
-	ctx interpolate.Context
+	ChefEnvironment            string               `mapstructure:"chef_environment"`
+	ChefLicense                string               `mapstructure:"chef_license"`
+	ConfigTemplate             string               `mapstructure:"config_template"`
+	CookbookPaths              []string             `mapstructure:"cookbook_paths"`
+	RolesPath                  string               `mapstructure:"roles_path"`
+	DataBagsPath               string               `mapstructure:"data_bags_path"`
+	EncryptedDataBagSecretPath string               `mapstructure:"encrypted_data_bag_secret_path"`
+	EnvironmentsPath           string               `mapstructure:"environments_path"`
+	ExecuteCommand             string               `mapstructure:"execute_command"`
+	InstallCommand             string               `mapstructure:"install_command"`
+	RemoteCookbookPaths        []string             `mapstructure:"remote_cookbook_paths"`
+	Json                       map[string]cty.Value `mapstructure:"json"`
```

Comment:
I wonder what happens here for JSON templates 🤔

Reply:
I'm still in the process of testing and writing the tests! I removed the wip but I found an issue and I'm working on it.